src/neo/IO/Caching/HashSetCache.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Neo.IO.Caching
+{
+    public class HashSetCache<T> : IReadOnlyCollection<T> where T : IEquatable<T>
+    {
+        private readonly int bucketCapacity;
+        private readonly int bucketCount;
+        private readonly LinkedList<HashSet<T>> sets = new LinkedList<HashSet<T>>();
+
+        public int Count
+        {
+            get
+            {
+                int count = 0;
+                foreach (var set in sets)
+                {
+                    count += set.Count;
+                }
+                return count;
+            }
+        }
+
+        public HashSetCache(int bucketCapacity, int bucketCount = 10)
+        {
+            if (bucketCapacity <= 0) throw new ArgumentOutOfRangeException(nameof(bucketCapacity));
+            if (bucketCount <= 0 || bucketCount > 20) throw new ArgumentOutOfRangeException($"{nameof(bucketCount)} should between 1 and 20");
```

Comment:
Why limit it to 20?

Reply:
It is not a strong demand, but I think bucketCount needs an upper limit, and personally think that bucketCount doesn't need to be too large, so limit to 20.  What's your opinion?

Reply:
I think we don't need a limit.

Reply:
Removing it is fine.